amulet/deployer.py

```diff
@@ -136,6 +136,15 @@ def add(self, service_name,
         if placement is not None:
             service['to'] = placement
 
+        if constraints is None and 'JUJU_TEST_CONSTRAINTS' in os.environ:
```

Comment:
@dannf sure, so if a charm author sets an architecture your settings won't take place. There should probably be two options here

```
JUJU_TEST_CONSTRAINTS
JUJU_TEST_EXTRA_CONSTRAINTS
````

The first one is likely what you want, it'll overwrite and takes priority for constraints regardless if the author has created supplied one or not. The second uses the logic you've defined now.

So replacing `JUJU_TEST_CONSTRAINTS` with `JUJU_TEST_EXTRA_CONSTRAINTS`, then adding another if block for if os.environ has `JUJU_TEST_CONSTRAINTS` and those overriding/setting constraints for the service.

@tvansteenburgh opinions?